httpbp/middlewares.go

```diff
@@ -246,3 +255,133 @@ func SupportedMethods(method string, additional ...string) Middleware {
 		}
 	}
 }
+
+// recoverPanic recovers from any panics, logs them, and sets the returned error
+// to a generic 500 error. recoverPanic is always the last middleware in the
+// middleware chain, so it is the first one when returning which lets the error
+// bubble up into other middlewares. Since it is always added to the middleware
+// chain is a specific position, it is not exported.
+func recoverPanic(name string, next HandlerFunc) HandlerFunc {
```

Comment:
This should be in a separate commit, I think you should cherry pick multiple commits and do "rebase and merge" rather than squashing them.

Reply:
Also, it looks like the other parts of the change that added this are missing from this PR, you probably want to cherry-pick 

91c387c4c9ba9f4d468cea5cbfa2533e94d0f323

and 

56f88638113fea6bf488b288a0eedfc593bf9673

Reply:
this is why we do merges 😅